src/inmanta/env.py

```diff
@@ -285,6 +269,20 @@ def notify_change(self) -> None:
         site.addsitedir(self.site_packages_dir)
         importlib.invalidate_caches()
 
+        if const.PLUGINS_PACKAGE in sys.modules:
+            mod = sys.modules[const.PLUGINS_PACKAGE]
+            if mod is not None:
+                # Make mypy happy
+                assert mod.__spec__.submodule_search_locations is not None
+                if self.site_packages_dir not in mod.__spec__.submodule_search_locations and os.path.exists(
+                    os.path.join(self.site_packages_dir, const.PLUGINS_PACKAGE)
+                ):
+                    # A V2 module was installed in this virtual environment, but the inmanta_plugins package was already
```

Comment:
I don't understand this.

How does this reloading affect other packages contributing to the namespace and why is a reload of the plugin package required?

Reply:
**This is required to cover the following scenario:**

* Two venvs are stacked on top of each other. The parent vebv contains the `inmanta-core` package and the subvenv is empty.
* The `inmanta_plugins` package gets loaded before a V2 module is installed in the subvenv. This way, the module object in `sys.modules`, doesn't have the site dir of the subvenv in its `submodule_search_locations`. This field caches where the loader should look for the namespace package that are part of the `inmanta_plugins` namespace.
* When a V2 module gets installed in the subvenv now, the loader will not find the newly installed V2 module, because it's not considering the site dir of the subvenv.

Since the compiler venv is removed now, this scenario can only occur in test cases. If you don't like this solution, we can also patch everything using the `conftest.py`. 

**Regarding the impact of the reloading:**

From the documentation I cannot determine any side-effects. I can add a test to verify this explicitly.

Reply:
I would document this in the code, otherwise I'm fine